### PR TITLE
 [SystemZ][z/OS] TXT records in the GOFF reader

### PR DESCRIPTION
**Pull request by ysyeda**
This PR adds handling for TXT records in the GOFF reader.

### CODE REVIEW

**Comment by ysyeda** [comment]
This PR is a copy of https://github.com/llvm/llvm-project/pull/74526, with a fix the unit test failure on macos.

**Comment by jh7370** [comment]
Is it practical to split this into two commits for review purposes, with the original PR's version followed by just the changes needed to fix the issue you ran into? I think it's going to be a pain to review the difference otherwise.

**Comment by ysyeda** [comment]
> Is it practical to split this into two commits for review purposes, with the original PR's version followed by just the changes needed to fix the issue you ran into? I think it's going to be a pain to review the difference otherwise.

Sure, I split it into 2 commits.

**Review by jh7370 — APPROVED** [review]
LGTM

File: llvm/unittests/Object/GOFFObjectFileTest.cpp [hunk 1]
```diff
@@ -502,3 +502,100 @@ TEST(GOFFObjectFileTest, InvalidERSymbolType) {
         FailedWithMessage("ESD record 1 has unknown Executable type 0x03"));
   }
 }
+
+TEST(GOFFObjectFileTest, TXTConstruct) {
+  char GOFFData[GOFF::RecordLength * 6] = {};
+
+  // HDR record.
+  GOFFData[0] = 0x03;
+  GOFFData[1] = 0xF0;
```
Comment (mstorsjo) [inline llvm/unittests/Object/GOFFObjectFileTest.cpp:511]: This causes lots of warnings when compiled with GCC, like these:
```
../unittests/Object/GOFFObjectFileTest.cpp:511:17: warning: overflow in conversion from ‘int’ to ‘char’ changes value from ‘240’ to ‘'\37777777760'’ [-Woverflow]
  511 |   GOFFData[1] = 0xF0;
      |                 ^~~~
```

**Pull request merged**